src/dotnet/Chat.Service/Chats.cs

```diff
@@ -680,24 +704,61 @@ public virtual async Task<Chat_CopyChatResult> OnCopyChat(Chat_CopyChat command,
-        // var newChatId = (ChatId)placeChatId;
-        // var contact = await Contacts.GetForChat(session, newChatId, cancellationToken).ConfigureAwait(false);
-        // Log.LogInformation("Contact for chat id '{ChatId}' is {Contact}", newChatId, contact);
-        // if (contact == null || contact.PlaceId.IsNone || !contact.IsStored()) {
-        //     var backendCmd2 = new ContactsBackend_MoveChatToPlace(chatId, placeId);
-        //     await Commander.Call(backendCmd2, true, cancellationToken).ConfigureAwait(false);
-        //     executed = true;
-        // }
-        //
         {
-            var backendCmd3 = new AccountsBackend_CopyChat(chatId, placeId, maxEntryId, correlationId);
+            var backendCmd3 = new AccountsBackend_CopyChat(sourceChatId, placeId, maxEntryId, correlationId);
             var hasChanges3 = await Commander.Call(backendCmd3, true, cancellationToken).ConfigureAwait(false);
             hasChanges |= hasChanges3;
         }
 
         Log.LogInformation("<- OnCopyChat({CorrelationId})", correlationId);
         return new Chat_CopyChatResult(hasChanges, hasErrors);
     }
+
+    public virtual async Task OnPublishCopiedChat(Chat_PublishCopiedChat command, CancellationToken cancellationToken)
+    {
+        if (Computed.IsInvalidating())
+            return; // It just spawns other commands, so nothing to do here
+
+        var (session, newChatId, sourceChatId) = command;
+
+        if (!newChatId.IsPlaceChat)
+            throw StandardError.Constraint("New chat id should be a place chat id.");
+
+        var localChatId = sourceChatId.IsPlaceChat ? sourceChatId.PlaceChatId.LocalChatId : sourceChatId.Id;
+        if (newChatId.PlaceChatId.LocalChatId != localChatId)
+            throw StandardError.Constraint("New chat is seems not a copy of provided source chat.");
+
+        var newChat = await Get(session, newChatId, cancellationToken).Require().ConfigureAwait(false);
+        if (!newChat.Rules.IsOwner())
+            throw StandardError.Constraint("You must be the Owner of this chat to perform this operation.");
+
+        var place = await Places.Get(session, newChat.Id.PlaceId, cancellationToken).Require().ConfigureAwait(false);
+        if (!place.Rules.IsOwner())
+            throw StandardError.Constraint("You must be a chat's place owner to perform this operation.");
+
+        var sourceChat = await Get(session, sourceChatId, cancellationToken).ConfigureAwait(false);
+        var copiedChat = await GetCopiedChat(session, newChatId, cancellationToken).ConfigureAwait(false);
+
+        if (sourceChat != null && newChat.IsPublic != sourceChat.IsPublic) {
+            var changeChatCmd = new Chats_Change(session,
+                newChat.Id,
+                newChat.Version,
+                Change.Update(new ChatDiff {
+                    IsPublic = sourceChat.IsPublic,
+                }));
+            await Commander.Call(changeChatCmd, true, cancellationToken).ConfigureAwait(false);
```

Comment:
Why do you use `outermost` commands there instead of processing within same transaction?

Reply:
I think I can call ChatsBackend commands with IsOutermost = false, but ContactsBackend's command I have to call with IsOutermost = true since it uses another db context. Due to using ContactsBackend's command  I decided to run all child commands with IsOutermost = true.

Reply:
Fixed